# HG changeset patch
# User alta88@fixall.com
# Date 1745509557 21600
# Parent  9f649540dfa0382e921f5643f342484aa3d0133c
Feed Subscribe dialog enhancements
- Make the <tree> rows look like folderPane rows
- remove messemger.css and folderPane.css dependencies

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.

diff --git a/mail/themes/shared/mail/feedSubscribe.css b/mail/themes/shared/mail/feedSubscribe.css
--- a/mail/themes/shared/mail/feedSubscribe.css
+++ b/mail/themes/shared/mail/feedSubscribe.css
@@ -1,8 +1,36 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+@import url("chrome://global/skin/global.css");
+@import url("chrome://messenger/skin/variables.css");
+@import url("chrome://messenger/skin/folderColors.css");
+@import url("chrome://messenger/skin/shared/list-container.css");
+
+:root {
+  --listbox-border-radius: 3px;
+  --sidebar-background: var(--sidebar-background-color, var(--foldertree-background));
+  @media not (prefers-contrast) {
+    --foldertree-background: light-dark(var(--color-gray-10), var(--color-gray-80));
+  }
+  @media (prefers-contrast) {
+    &:not([lwtheme]) {
+      --foldertree-background: -moz-Dialog;
+    }
+  }
+}
+
+html,
+body {
+  height: 100%;
+}
+
+body {
+  display: flex;
+  flex-direction: column;
+}
+
 #feedSubscriptions {
   min-width: 20em;
   min-height: 40em;
 }
@@ -16,41 +44,108 @@
 #learnMore {
   align-self: end;
 }
 
+/* Tree styling - match folderPane */
 #folderTree {
   margin: 0;
+  background-color: var(--sidebar-background);
+}
+
+#feedTypeCol {
+  width: 138px;
 }
 
-treechildren::-moz-tree-image {
-  width: 16px;
-  height: 16px;
-  -moz-context-properties: fill, fill-opacity, stroke;
+tree {
+  & > treechildren::-moz-tree-row {
+    background-color: inherit;
+    border: 1px solid transparent;
+    border-radius: var(--listbox-border-radius);
+    margin-inline: 6px;
+  }
+
+  & > treechildren::-moz-tree-row(hover) {
+    border-color: var(--list-container-border-hover);
+    background-color: var(--list-container-background-hover);
+
+    @media (forced-colors) {
+      color: var(--selected-item-text-color);
+    }
+  }
+  & > treechildren::-moz-tree-row(selected, current) {
+    border-color: var(--list-container-border-selected-current);
+    background-color: var(--list-container-background-selected-current);
+  }
+  & > treechildren::-moz-tree-row(selected, current, focus) {
+    border-color: var(--list-container-border-selected-current-focused);
+    background-color: var(--list-container-background-selected-current-focused);
+    outline: none;
+  }
+
+  & > treechildren::-moz-tree-cell-text,
+  & > treechildren::-moz-tree-cell-text(selected) {
+    padding-inline-start: 3px;
+    color: currentColor;
+  }
+
+  /* No new messages star on tree-image since an svg background-image in
+   * treechildren does not recognize stroke/fill - would need to bring back
+   * the old .png image.*/
+  & > treechildren::-moz-tree-cell-text(folderNameCol, isServer-true),
+  & > treechildren::-moz-tree-cell-text(folderNameCol, newMessages-true) {
+    font-weight: bold;
+  }
+
+  & > treechildren::-moz-tree-cell-text(folderNameCol, isServer-true, biffState-NewMail),
+  & > treechildren::-moz-tree-cell-text(folderNameCol, newMessages-true) {
+    color: var(--new-folder-color);
+  }
+
+  & > treechildren::-moz-tree-image {
+    width: 16px;
+    height: 16px;
+    margin-inline-end: 2px;
+    -moz-context-properties: fill, fill-opacity, stroke;
+  }
+
+  & > treechildren::-moz-tree-twisty {
+    color: currentColor;
+    padding-inline-start: 8px;
+  }
 }
 
-/* An svg background-image in treechildren does not recognize stroke/fill. */
-treechildren::-moz-tree-cell-text(folderNameCol, newMessages-true) {
-  background: none !important;
+/* ..... Default folder ..... */
+treechildren::-moz-tree-image(folderNameCol) {
+  list-style-image: var(--folder-pane-folder);
+  fill: color-mix(in srgb, var(--folder-color-folder) 20%, transparent);
+  stroke: var(--folder-color-folder);
 }
 /* ..... Feed server/account ..... */
 treechildren::-moz-tree-image(folderNameCol, isServer-true, serverType-rss) {
+  --icon-color: var(--primary);
   list-style-image: var(--folder-pane-rss);
-  fill: var(--primary-fill);
-  stroke: var(--primary-stroke);
+  fill: color-mix(in srgb, var(--icon-color, var(--folder-color-folder)) 20%, transparent);
+  stroke: var(--icon-color, var(--folder-color-folder));
 }
 /* ..... Feed folder ..... */
 treechildren::-moz-tree-image(folderNameCol, isFeedFolder-true) {
   list-style-image: var(--folder-pane-folder-rss);
   fill: color-mix(in srgb, var(--folder-color-folder-rss) 20%, transparent);
-  stroke: var(--folder-color-folder-rss) !important;
+  stroke: var(--folder-color-folder-rss);
 }
 /* ..... Feed subscription item ..... */
 treechildren::-moz-tree-image(feedTypeCol, isFeed-true),
 treechildren::-moz-tree-image(folderNameCol, isFeed-true) {
   list-style-image: var(--folder-pane-rss);
   fill: color-mix(in srgb, var(--folder-color-rss) 20%, transparent);
   stroke: var(--folder-color-rss);
 }
+/* ..... Archives ..... */
+treechildren::-moz-tree-image(folderNameCol, specialFolder-Archive) {
+  list-style-image: var(--folder-pane-archive);
+  fill: color-mix(in srgb, var(--folder-color-archive) 20%, transparent);
+  stroke: var(--folder-color-archive);
+}
 
 treechildren::-moz-tree-image(feedTypeCol, isFeed-true, isAtomFeed) {
   stroke: MediumAquamarine; /* Atom feed color from somewhere */
 }
@@ -73,12 +168,9 @@ treechildren::-moz-tree-image(feedTypeCo
   list-style-image: var(--icon-warning);
   fill: var(--color-red-60);
   stroke: var(--color-amber-30);
 }
-
-#feedTypeCol {
-  width: 138px;
-}
+/* End tree styling */
 
 a[disabled],
 fieldset:disabled *,
 label:has(:disabled) {
@@ -127,9 +219,12 @@ label:has(:disabled) {
   text-overflow: ellipsis;
   flex: 1;
 }
 
-#locationBox, #folderBox, #statusBox, #actionButtonsBox {
+#locationBox,
+#folderBox,
+#statusBox,
+#actionButtonsBox {
   display: flex;
   align-items: center;
   /* No ellipsis without this! */
   min-width: 0;
@@ -158,7 +253,14 @@ label:has(:disabled) {
   margin: 0;
   padding: 0;
 }
 
+input[type="number"] {
+  margin: 2px 4px;
+  &.size3 {
+    width: calc(3ch + 44px);
+  }
+}
+
 #autotagPrefix {
   max-width: 15em;
 }
diff --git a/mailnews/extensions/newsblog/feed-subscriptions.xhtml b/mailnews/extensions/newsblog/feed-subscriptions.xhtml
--- a/mailnews/extensions/newsblog/feed-subscriptions.xhtml
+++ b/mailnews/extensions/newsblog/feed-subscriptions.xhtml
@@ -23,10 +23,8 @@
     <meta
       http-equiv="Content-Security-Policy"
       content="default-src chrome:; script-src chrome: 'unsafe-inline'; style-src chrome: 'unsafe-inline'; img-src chrome: data:"
     />
-    <link rel="stylesheet" href="chrome://messenger/skin/messenger.css" />
-    <link rel="stylesheet" href="chrome://messenger/skin/folderPane.css" />
     <link rel="stylesheet" href="chrome://messenger/skin/icons.css" />
     <link rel="stylesheet" href="chrome://messenger/skin/folderMenus.css" />
     <link
       rel="stylesheet"
